src/components/Bag/classes.js

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import { v4 as uuidv4 } from "uuid";
+import { Label, InsertDriveFile, Folder } from "@material-ui/icons";
+
+class BagItem {
+    constructor(item) {
+        this.item = item;
+    }
+
+    icon(t) {
+        return <Label />;
+    }
+
+    get label() {
+        return (
+            this.item?.label ||
+            this.item?.name ||
+            this.item?.description ||
+            "override label() please"
+        );
+    }
+
+    get shareable() {
+        return false;
+    }
+
+    get downloadable() {
+        return false;
+    }
+
+    get id() {
+        return this.item.id;
+    }
+}
+
+class FileBagItem extends BagItem {
+    icon(t) {
+        return <InsertDriveFile />;
+    }
+
+    get shareable() {
+        return true;
+    }
+
+    get downloadable() {
+        return true;
+    }
+
+    get name() {
+        return this.item.path.substring(this.item.path.lastIndexOf("/") + 1);
+    }
+
+    get path() {
+        return this.item.path;
+    }
+}
+
+class FolderBagItem extends BagItem {
+    icon(t) {
+        return <Folder />;
+    }
+
+    get shareable() {
+        return true;
+    }
+
+    get name() {
+        return this.item.path.substring(this.item.path.lastIndexOf("/") + 1);
+    }
+    get path() {
+        return this.item.path;
+    }
+}
+
+class AnalysisBagItem extends BagItem {
+    icon(t) {
+        return <img src="/analyses_selected.png" alt={t("common:analyses")} />;
+    }
+
+    get shareable() {
+        return true;
+    }
+}
+
+class AppBagItem extends BagItem {
```

Comment:
I may have missed it, but at least for sharing, app items also need to contain the `system_id`

Reply:
The nested item attribute in the AppBagItem contains the unaltered object from the app listing it originated from, and that includes the system_id. The item object gets pulled up as the main object before getting handed off to the sharing dialog.